### PR TITLE
Add synctable command into caravel

### PR DESCRIPTION
**Pull request by linroex**
Sometime we have many table need insert to caravel tables, I write a command can help people to do this, set a table name prefix or default is data_ , this cmd will scan caravel has table and database all table, compare and insert it.

### CODE REVIEW

File: caravel/bin/caravel [hunk 1]
```diff
@@ -125,6 +125,34 @@ def refresh_druid():
...
+    ''' Sync DB Table with Caravel Table'''
+    exists_tables = []
+    for row in db.session.query(caravel.models.SqlaTable).all():
+        exists_tables += [(row.database, row.name)]
+    
+    all_tables = []
+    for _db in db.session.query(caravel.models.Database).all():
+        for _tablename in _db.get_sqla_engine().table_names():
+            all_tables += [(_db.database_name, _tablename)]
+    all_tables = [_ for _ in all_tables if _[1].startswith(prefix)]
```
Comment (xrmx) [inline caravel/bin/caravel]: use something like table instead of `_`,  as it usually used to discard stuff

File: caravel/bin/caravel [hunk 2]
```diff
@@ -125,6 +125,34 @@ def refresh_druid():
             "[" + cluster.cluster_name + "]")
     session.commit()
 
+@manager.option(
+    '-p', '--prefix', default='data_',
+    help="Sync Table Prefix")
+def synctable(prefix):
+    ''' Sync DB Table with Caravel Table'''
+    exists_tables = []
```
Comment (xrmx) [inline caravel/bin/caravel]: existing_tables

File: caravel/bin/caravel [hunk 3]
```diff
@@ -125,6 +125,34 @@ def refresh_druid():
...
+    
+    all_tables = []
+    for _db in db.session.query(caravel.models.Database).all():
+        for _tablename in _db.get_sqla_engine().table_names():
+            all_tables += [(_db.database_name, _tablename)]
+    all_tables = [_ for _ in all_tables if _[1].startswith(prefix)]
+    
+    need_insert = list(set(all_tables) - set(exists_tables))
+
+    for row in need_insert:
```
Comment (xrmx) [inline caravel/bin/caravel]: for db_name, table_name in need_insert:

File: caravel/bin/caravel [hunk 4]
```diff
@@ -125,6 +125,34 @@ def refresh_druid():
             "[" + cluster.cluster_name + "]")
     session.commit()
 
+@manager.option(
+    '-p', '--prefix', default='data_',
```
Comment (xrmx) [inline caravel/bin/caravel]: I'd make the prefix not required and empty by default so one can insert all the tables of a database if needed.

File: caravel/bin/caravel [hunk 5]
```diff
@@ -125,6 +125,34 @@ def refresh_druid():
...
+    need_insert = list(set(all_tables) - set(exists_tables))
+
+    for row in need_insert:
+        db_name = row[0]
+        table_name = row[1]
+
+        tbl = caravel.models.SqlaTable(table_name=table_name)
+        tbl.description = ""
+        tbl.is_featured = False
+        tbl.database = db.session.query(caravel.models.Database).filter_by(database_name=db_name).first()
```
Comment (xrmx) [inline caravel/bin/caravel]: i think you can query this once outside the loop

File: caravel/bin/caravel [hunk 6]
```diff
@@ -125,6 +125,34 @@ def refresh_druid():
...
+        db.session.merge(tbl)
```
Comment (xrmx) [inline caravel/bin/caravel]: db.session.add(tbl) should be enough

File: caravel/bin/caravel (hunk 5 above)
Comment (linroex) [inline caravel/bin/caravel]: You mean #152 ?
It is not same in each time loop

File: caravel/bin/caravel (hunk 4 above)
Comment (linroex) [inline caravel/bin/caravel]: I think it is not good because some table in caravel db store user,permission...etc data, they don't need insert to caravel.

**Comment by linroex** [comment]
I fix some bug, so close this pr and create another one

File: caravel/bin/caravel (hunk 4 above)
Comment (xrmx) [inline caravel/bin/caravel]: Yeah but 'data_' is completely arbitrary, if you don't want to read all the tables fine, just don't add a default that would work only for you :)

**Pull request closed without merging**